File: src/main/java/bad/robot/radiate/ui/SwitchTo.java
package bad.robot.radiate.ui;

import java.awt.*;
import java.awt.event.AWTEventListener;
import java.awt.event.KeyAdapter;
import java.awt.event.KeyEvent;
import java.util.function.Supplier;

import static bad.robot.radiate.Main.Radiate;

@Deprecated
class SwitchTo extends KeyAdapter implements AWTEventListener {

    private final Supplier<FrameFactory> mode;
    private final int keyCode;

    public SwitchTo(Supplier<FrameFactory> mode, int keyCode) {
        this.mode = mode;
        this.keyCode = keyCode;
    }

    @Override
    public void keyPressed(KeyEvent event) {
        if (event.getKeyCode() == keyCode) {
            Radiate.stop();
            Radiate.start(Radiate.getCurrentTasks(), mode.get());
        }
    }

    @Override
    public void eventDispatched(AWTEvent event) {
        if (event.getID() == KeyEvent.KEY_PRESSED)
            keyPressed((KeyEvent) event);
    }

}

File: src/main/scala/bad/robot/radiate/ui/SwitchTo.scala
package bad.robot.radiate.ui

import java.awt._
import java.awt.event.{AWTEventListener, KeyAdapter, KeyEvent}

import bad.robot.radiate.Main.Radiate

class SwitchTo(mode: => FrameFactory, keyCode: Int) extends KeyAdapter with AWTEventListener {

  override def keyPressed(event: KeyEvent) {
    if (event.getKeyCode == keyCode) {
      Radiate.stop()
      Radiate.start(Radiate.getCurrentTasks, mode)
    }
  }

  def eventDispatched(event: AWTEvent) {
    if (event.getID == KeyEvent.KEY_PRESSED) keyPressed(event.asInstanceOf[KeyEvent])
  }
}